### PR TITLE
feat(secure): Add Malware, Drift, ML and AWS ML policy resources

### PR DESCRIPTION
**Pull request by alexeyovriakh**
## Summary

- This PR adds support for Malware, Drift, ML and AWS ML policies.
- See the screenshots below for the **draft** TF resource design (the final design might deviate).

## Testing steps

1. Run `make install` in local `terraform-provider-sysdig` directory
2. Create a new directory for customer-side TF script
3. Create a new file called `main.tf` in that directory
4. Run `terraform init` (precede with `rm .terraform.lock.hcl` if you've changed the code), update the values if needed, and run `terraform apply`. Enter "yes" to confirm the changes. 
5. You should see a successful message

```terraform
// main.tf
terraform {
  required_providers {
    sysdig = {
      source  = "local/sysdiglabs/sysdig"
      version = "~> 1.0.0"
    }
  }
}

provider "sysdig" {
  sysdig_secure_url          = "https://secure-staging.sysdig.com"
  sysdig_secure_api_token    = "<TOKEN>"
  sysdig_secure_insecure_tls = true
}

resource "sysdig_secure_notification_channel_email" "email" {
  name       = "Test Email Channel 14"
  recipients = ["alexey.ovriakh@sysdig.com"]
}

data "sysdig_secure_malware_policy" "p" {
  name = "Test Malware Policy 9"
}

resource "sysdig_secure_malware_policy" "test_policy" {
  name        = "Test Malware Policy 14"
  description = "Test Malware Policy Description"
  enabled     = true
  severity    = 4

  rules {
    enabled = true
    description = "Test Malware Rule Description"
    tags        = ["tag1", "tag2"]

    details {
      use_managed_hashes = true

      additional_hashes {
        hash         = "304ef4cdda3463b24bf53f9cdd69ad3ecdab0842e7e70e2f3cfbb9f14e1c4ae6"
        hash_aliases = ["hash_name", "hash_name_2"]
      }

      ignore_hashes {
        hash         = "6ac3c336e4094835293a3fed8a4b5fedde1b5e2626d9838fed50693bba00af0e"
        hash_aliases = ["ignore_hashes", "ignore_hashes_2"]
      }
    }
  }
  
  actions {
    prevent_malware = true
    container = "stop"
    # capture
  }

  notification_channels = [sysdig_secure_notification_channel_email.email.id]
}

output "sysdig_secure_malware_policy_1" {
  value = sysdig_secure_malware_policy.test_policy
}

output "sysdig_secure_malware_policy_x" {
  value = data.sysdig_secure_malware_policy.p
}

```

## Draft design 
Screenshot 2024-01-26 at 15 59 54
Screenshot 2024-01-26 at 15 57 03
Screenshot 2024-01-26 at 15 56 54
Screenshot 2024-01-26 at 15 56 45

### CODE REVIEW

File: sysdig/data_source_sysdig_secure_composite_policy.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		"rules": {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"id":          ReadOnlyIntSchema(),
+					"name":        ReadOnlyStringSchema(),
+					"enabled":     EnabledComputedSchema(),
+					"description": DescriptionComputedSchema(),
+					"tags":        TagsComputedSchema(),
```
Comment (kmvachhani) [inline sysdig/data_source_sysdig_secure_composite_policy.go:61]: we don't have a concept of description/tags on malware rules (and i think all other composite policy types), did you find any that needed these fields?

Comment (kmvachhani) [reply sysdig/data_source_sysdig_secure_composite_policy.go:61]: actually looks like we have tags but by default we set it to just "malware"

Comment (kmvachhani) [reply sysdig/data_source_sysdig_secure_composite_policy.go:61]: and looks like rule description is just the policy name + `malware rule` when made through the UI. @ombellare do you think it's okay to let customers customize these 2 fields?

File: sysdig/data_source_sysdig_secure_composite_policy.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		"notification_channels": NotificationChannelsComputedSchema(),
+		"runbook":               RunbookComputedSchema(),
+		"rules": {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"id":          ReadOnlyIntSchema(),
+					"name":        ReadOnlyStringSchema(),
+					"enabled":     EnabledComputedSchema(),
```
Comment (kmvachhani) [inline sysdig/data_source_sysdig_secure_composite_policy.go:59]: i think we should only keep `enabled` at the policy level

File: sysdig/data_source_sysdig_secure_composite_policy.go (hunk 1 above)
Comment (alexeyovriakh) [reply sysdig/data_source_sysdig_secure_composite_policy.go]: Thanks @kmvachhani. We caught up with @ivanlysiuk-sysdig to clarify the changes. Tags have been hidden.

**Review by kmvachhani — APPROVED** [review]

**Review by ivanlysiuk-sysdig — APPROVED** [review]

**Review by ombellare — COMMENTED** [review]
Sorry for the late review, still checking some files

File: sysdig/schema.go [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"enabled": {
+					Type:     schema.TypeBool,
+					Computed: true,
+				},
+				"threshold": {
+					Type:     schema.TypeInt,
```
[the remark this answers is not in the document]
Comment (ombellare) [reply sysdig/schema.go:410]: Or probably keep it as 1, 2, 3 but in the doc mention what they mean internally

**Pull request merged**